x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java

```diff
@@ -271,7 +282,32 @@ public void onIndexModule(IndexModule indexModule) {
         if (SearchableSnapshotsConstants.isSearchableSnapshotStore(indexSettings.getSettings())
             && indexSettings.getSettings().getAsBoolean("index.frozen", false) == false) {
             return Optional.of(
-                engineConfig -> new ReadOnlyEngine(engineConfig, null, new TranslogStats(), false, Function.identity(), false)
+                engineConfig -> new ReadOnlyEngine(engineConfig, null, new TranslogStats(), false, Function.identity(), false) {
+
+                    // present an empty IndexCommit to the snapshot mechanism so that we copy no shard data to the repository
+                    private final IndexCommit emptyIndexCommit;
+
+                    {
+                        try {
+                            final Directory directory = engineConfig.getStore().directory();
+                            final String oldestSegmentsFile = Arrays.stream(directory.listAll())
+                                .filter(s -> s.startsWith(IndexFileNames.SEGMENTS + "_"))
+                                .min(Comparator.naturalOrder())
```

Comment:
Oops, this will almost always work except for cases where we add another character to the encoded generation (think `"1" < "10" < "2"`). I'll address this.